### PR TITLE
Add javadoc to UniAsserter

### PR DESCRIPTION
**Pull request by geoand**
Relates to: #29617

### CODE REVIEW

File: test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java [hunk 1]
```diff
@@ -6,53 +6,124 @@
 
 import io.smallrye.mutiny.Uni;
 
+/**
+ * Used in conjunction with {@link RunOnVertxContext} (or similar annotations) in order to test assertions
+ * on APIs that return {@link Uni} and whose execution needs to occur on a Vert.x Event Loop Thread (such as Hibernate Reactive
+ * for example).
+ *
+ * The following example shows how this class is meant to be used:
```
Comment (mkouba) [inline test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java:14]: Shouldn't we explicitly state that it must be used as a parameter of a test method?

Comment (geoand) [reply test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java:14]: Good point

**Review by mkouba — COMMENTED** [review]
Added one minor comment. Otherwise looks good!

File: test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java (hunk 1 above)
Comment (geoand) [reply test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java:14]: Should be better now

**Review by mkouba — APPROVED** [review]

**Pull request merged**